classes/UpgradeTools/FileFilter.php

```diff
@@ -160,9 +161,15 @@ public function getFilesToIgnoreOnUpgrade()
                 if (!$fileinfo->isDir() || $fileinfo->isDot()) {
                     continue;
                 }
-                if (in_array($fileinfo->getFilename(), $nativeModules)) {
-                    $this->excludeAbsoluteFilesFromUpgrade[] = '/modules/' . $fileinfo->getFilename();
+                if (!in_array($fileinfo->getFilename(), $nativeModules)) {
+                    continue;
+                }
+                if (!(new SplFileInfo($this->rootDir . '/modules/' . $fileinfo->getFilename() . '/vendor'))->isDir()) {
+                    // If a vendor folder is found in the module, this means it has been upgraded or manually installed
+                    // and can be ignored during the upgrade process
```

Comment:
Why ? If we don't have vendor, how the module can work properly ?

Reply:
Module that are provided with a PrestaShop release have been added with composer, meaning their dependencies are not stored in their own vendor /folder, they are merged the core one.
It works because:
* The PrestaShop autoloader is already loaded and module files are found in it
* We check before trying to load the module autoloader, it may not exist https://github.com/PrestaShop/ps_googleanalytics/blob/618ade8f9a424ad31968f6367ddf3e0be8c5f879/ps_googleanalytics.php#L24-L27

Reply:
Okay I didn't understand that it was only for native modules. Thank you for the explanation.